### PR TITLE
Initial rough spaces implementation - removed the concept of a rooms …

### LINKED ISSUE #70: Add Spaces Support
Organization is goood

### PR DESCRIPTION
**Pull request by Limvot**
…list, now everything is like a chat room, since that's what spaces are anyway, and the room list is just a fake space. ~~Not currently sorting or anything, but~~ the links work, and it builds a room_id path that you can back out of.

Closes #70 .

Now properly handles sorting, has an All Rooms list, displays live previews of all rooms from room selection, properly aggregates notifications for spaces from children, hierarchy finds most recent event in each space and sorts accordingly, etc.

This is about? More or less? ready to be reviewed and merged. I def want some solid reviews before merging this, but it's not a draft anymore!

Note this also already depends on pinned_messages and rich replies.

### CODE REVIEW

File: serif_swing/src/main/kotlin/serif_swing/App.kt [hunk 1]
```diff
@@ -1147,7 +1078,7 @@ class App {
         if (partial) {
             when {
                 new_state is MatrixChatRoom && s is SwingChatRoom -> { s.update(new_state, frame.width); return; }
-                new_state is MatrixRooms && s is SwingRooms -> { s.update(new_state); return; }
+                //new_state is MatrixRooms && s is SwingRooms -> { s.update(new_state); return; }
```
Comment (lotusronin) [inline serif_swing/src/main/kotlin/serif_swing/App.kt]: Remove this comment

File: serif_swing/src/main/kotlin/serif_swing/App.kt [hunk 2]
```diff
@@ -1162,7 +1093,7 @@ class App {
                 { javax.swing.SwingUtilities.invokeLater({ refresh_all() }) },
                 panel, mstate
             )
-            is MatrixRooms -> SwingRooms(::transition, panel, mstate)
+            //is MatrixRooms -> SwingRooms(::transition, panel, mstate)
```
Comment (lotusronin) [inline serif_swing/src/main/kotlin/serif_swing/App.kt]: Remove this comment

**Review by lotusronin — CHANGES_REQUESTED** [review]
Remove unnecessary UI elements for the space view. Would also like to deduplicate the unread message counts.

**Review by enoxone — APPROVED** [review]

**Review by lotusronin — APPROVED** [review]
lgtm :+1:

**Review by lotusronin — APPROVED** [review]
new cache looks good!

**Pull request merged**